zxcvbn-test/ZxcvbnTestMesserliExtended.cs

```diff
@@ -0,0 +1,138 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.IO;
+using System.IO.Compression;
+
+namespace zxcvbn_test
+{
+    [TestClass]
+    public class ZxcvbnTest_MesserliExtended
+    {
+        private const string CategoryUnitTest = "UnitTest";
+
+        [TestMethod]
+        [TestCategory(CategoryUnitTest)]
+        [Priority(1)]
+        public void ZxcvbnEmptyPasswordMetric()
+        {
+            var passwordChecker = new Zxcvbn.Zxcvbn();
+
+            var result = passwordChecker.EvaluatePassword("");
+
+            Assert.AreEqual(0.0, result.Entropy);
+        }
+
+        [TestMethod]
+        [TestCategory(CategoryUnitTest)]
+        [Priority(1)]
+        public void ZxcvbnPasswordMetric()
+        {
+            var passwordChecker = new Zxcvbn.Zxcvbn();
+
+            var a = passwordChecker.EvaluatePassword("a");
+            var aaaa = passwordChecker.EvaluatePassword("aaaa");
+            var abab = passwordChecker.EvaluatePassword("abab");
+            var ABab = passwordChecker.EvaluatePassword("ABab");
+
+            Assert.IsTrue(aaaa.Entropy > a.Entropy);
+            Assert.IsTrue(abab.Entropy > aaaa.Entropy);
+            Assert.IsTrue(ABab.Entropy > abab.Entropy);
+        }
+
+        [TestMethod]
+        [TestCategory(CategoryUnitTest)]
+        [Priority(1)]
+        public void DictionaryMatcherTest()
+        {
+            // We test if the dictionary matcher loads the messerli dictionary
+            var passwordChecker = new Zxcvbn.Zxcvbn();
+            var password = "Messerli";
+            var metric = passwordChecker.EvaluatePassword(password);
+
+            foreach (var match in metric.MatchSequence)
+            {
+                // Match
+                Assert.AreEqual(10, match.Cardinality);
+                Assert.AreEqual(password, match.Token);
+                Assert.AreEqual(0, match.i);
+                Assert.AreEqual(password.Length - 1, match.j);
+
+                // Dictionary Match
+                var dictionaryMatch = match as Zxcvbn.Matcher.DictionaryMatch;
+
+                Assert.AreNotEqual(null, dictionaryMatch);
+                Assert.AreEqual("messerli", dictionaryMatch.DictionaryName);
+                Assert.AreEqual(1, dictionaryMatch.UppercaseEntropy);
+
+
+            }
+        }
+
+        [TestMethod]
+        [TestCategory(CategoryUnitTest)]
+        [Priority(1)]
+        public void SwissSpatialMatcherTest()
+        {
+            // We test if the pattern matcher matches swiss keyboards
```

Comment:
Useless comment

Reply:
Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/533e3068b1feab4aa0a4f6866519c1847a6da367